addons/survey/controllers/main.py

```diff
@@ -302,41 +302,53 @@ def survey_submit(self, survey_token, answer_token, **post):
         # Survey Validation
         access_data = self._get_access_data(survey_token, answer_token, ensure_token=True)
         if access_data['validity_code'] is not True:
-            return {
-                'error': access_data['validity_code'],
-            }
+            return {'error': access_data['validity_code']}
         survey_sudo, answer_sudo = access_data['survey_sudo'], access_data['answer_sudo']
 
+        if answer_sudo.state == 'done':
+            return {'error': _('Unauthorized')}
```

Comment:
Commit1: as returned error are keys, it is strange to have a translated key. Moreover is it not really used, but as for other error keys, it is only a generic "there was an error", is it intended ?

Reply:
Indeed, for some reason I thought the JS used the text, I've put back regular error keys instead.